src/sagas/pushNotification.js

```diff
@@ -444,6 +445,28 @@ export function* dismissOptInQuestion() {
   yield STORE.setItem(pushNotificationKey.optInDismissed, true);
 }
 
+const cleanToken = async () => {
+  await messaging().unregisterDeviceForRemoteMessages();
+  await messaging().deleteToken();
+};
+
+/**
+ * This function is responsible for reset the push notification.
+ */
+export function* resetPushNotification() {
+  // Unregister the device from FCM
+  yield call(cleanToken);
+  // Clean the store
+  yield STORE.removeItem(pushNotificationKey.enabledAt);
+  yield STORE.removeItem(pushNotificationKey.settings);
+  yield STORE.removeItem(pushNotificationKey.hasBeenEnabled);
+  yield STORE.removeItem(pushNotificationKey.optInDismissed);
+  yield STORE.removeItem(pushNotificationKey.deviceId);
+  // Reset the state
+  yield put(pushReset());
+  yield fork(onAppInitialization);
```

Comment:
Why do you need this in the end?

Reply:
Also confused about what was the idea here

`onAppInitialization` calls `take` on `onStartWalletSuccess`, and I guess that at this point, it will not get called again, right?

Also, if you want this to run as a process in parallel, you should use `spawn` instead of `fork` as `fork` will spawn an attached process

Reply:
I have changed this process in the feature toggle PR [a94016f](https://github.com/HathorNetwork/hathor-wallet-mobile/pull/240/commits/a94016f451a41be028b079b572fc5ce6e5c5e411#diff-b2f41e02468f99f5c9c39db9c11e7c79fbde4091cf522468d84af47845be7adaR247)

Reply:
So this is wrong? Why no change here? If it does not work as expected we shouldn't merge.

Reply:
I realized it only after I was working on the feature toggle. As the PRs are chained, when I merge the feature toggle PR here, it will be resolved. That's why I didn't solve it here.

Reply:
Removed.